sdk/test/integration/api_trace_test.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::SDK, 'API_trace' do
+  let(:sdk) { OpenTelemetry::SDK }
+  let(:exporter) { sdk::Trace::Export::InMemorySpanExporter.new }
+  let(:span_processor) { sdk::Trace::Export::SimpleSpanProcessor.new(exporter) }
+  let(:factory) do
+    OpenTelemetry.tracer_factory = sdk::Trace::TracerFactory.new.tap do |factory|
+      factory.add_span_processor(span_processor)
+    end
+  end
+  let(:tracer) { factory.tracer(__FILE__, sdk::VERSION) }
+  let(:remote_span_context) do
+    OpenTelemetry::Trace::SpanContext.new(remote: true)
+  end
+
+  describe 'tracing root spans' do
+    before do
+      @root_span = tracer.start_root_span('root')
+
+      tracer.with_span(@root_span) do
+        tracer.in_span('child of root') do |span|
+          @child_of_root = span
+        end
+      end
+    end
+
+    it 'traces root spans' do
+      _(@root_span.name).must_equal 'root'
+    end
+
+    it 'root has a child' do
+      _(@child_of_root.to_span_data.parent_span_id).must_equal @root_span.to_span_data.span_id
+    end
+
+    it 'root has accurate child_count' do
+      # TODO: is child_count maintained by an exporter or processor?
+      _(@root_span.to_span_data.child_count).must_equal 1
+    end
+
+    it 'root has accurate total_recorded_links' do
+      _(@root_span.to_span_data.total_recorded_links).must_equal 0
+    end
+
+    it "doesn't have links" do
+      assert_nil @child_of_root.links
+    end
+  end
+
+  describe 'tracing child-of-remote spans' do
+    before do
+      @remote_span = tracer.start_span('remote', with_parent_context: remote_span_context)
+      @child_of_remote = tracer.start_span('child1', with_parent: @remote_span)
```

Comment:
Generally speaking, `child_of_remote` should be created using `with_parent_context: @remote_span.context` because we won't have a direct reference to the remote parent (the parent Span exists as an object in the remote process, but in the local process it only exists as a SpanContext).

Reply:
Changed :heavy_check_mark: